### PR TITLE
Adds support for SPE Ring / x-ms-clitelem parsing for /token requests

### PR DESCRIPTION
**Pull request by iambmelt**
Work in progress PR.

This PR contains formatting fixes, import scrubbing.

Implements support for SPE Ring/x-ms-clitelem header parsing. The result is set to the `TokenResult` and later packaged up for broker consumers to parse out.

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/TokenCommand.java [hunk 1]
```diff
@@ -76,28 +76,41 @@ public TokenCommand(@NonNull final Context context,
...
+                                    + ": Succeeded"
                     );
+
                     return result;
                 }
-            }catch(UiRequiredException e){
-                if(e.getErrorCode().equals(UiRequiredException.INVALID_GRANT)){
+            } catch (UiRequiredException e) {
+                if (e.getErrorCode().equals(UiRequiredException.INVALID_GRANT) // was invalid_grant
+                        && mControllers.size() > ii + 1) { // isn't the last controller we can try
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/TokenCommand.java]: 👍

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenResponse.java [hunk 2]
```diff
@@ -54,6 +54,98 @@
...
+    public void setRefreshTokenAge(String refreshTokenAge) {
+        this.mRefreshTokenAge = refreshTokenAge;
+    }
+
+    /**
+     * Gets the server error code.
+     *
+     * @return The server error code to get.
+     */
+    public String getServerErrorCode() {
```
Comment (heidijinxujia) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenResponse.java]: Q: should these be in MicrosoftTokenErrorResponse?

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerTokenResponse.java [hunk 3]
```diff
@@ -62,10 +62,14 @@ public BrokerTokenResponse(MicrosoftStsTokenResponse microsoftStsTokenResponse){
         setClientId(microsoftStsTokenResponse.getClientId());
         setExtExpiresIn(microsoftStsTokenResponse.getExtExpiresIn());
         setFamilyId(microsoftStsTokenResponse.getFamilyId());
+        setSpeRing(microsoftStsTokenResponse.getSpeRing());
+        setRefreshTokenAge(microsoftStsTokenResponse.getRefreshTokenAge());
+        setServerErrorCode(microsoftStsTokenResponse.getServerErrorCode());
```
Comment (heidijinxujia) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerTokenResponse.java]: Q: 
Should these
`setServerErrorCode(microsoftStsTokenResponse.getServerErrorCode());
setServerSubErrorCode(microsoftStsTokenResponse.getServerSubErrorCode());`
move to MicrosoftTokenErrorResponse?

The token result contains two properties
1. TokenResponse for the successful response
 2. TokenErrorResponse for the error response

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenResponse.java (hunk 2 above)
Comment (heidijinxujia) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenResponse.java]: Should these also be in part of MicrosoftTokenErrorResponse?

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/TokenCommand.java (hunk 1 above)
Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/controllers/TokenCommand.java]: Resolves https://github.com/AzureAD/azure-activedirectory-library-for-android/issues/1394

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerTokenResponse.java (hunk 3 above)
Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerTokenResponse.java]: Synced offline. This code just ensures that data isn't "lost" when parceling

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenResponse.java (hunk 2 above)
Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenResponse.java]: Synced offline - the result is moved by the `LocalAuthenticationResult` or via `BaseException` (if things go wrong)

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenResponse.java [hunk 4]
```diff
@@ -54,6 +54,98 @@
...
+
+    /**
+     * The age of the RT, according to the server.
+     */
+    private String mRefreshTokenAge;
+
+    /**
+     * The error code surfaced by the server.
+     */
+    private String mServerErrorCode;
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenResponse.java]: Consider renaming this.

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenResponse.java [hunk 5]
```diff
@@ -54,6 +54,98 @@
...
+
+    /**
+     * The server suberror code.
+     */
+    private String mServerSubErrorCode;
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenResponse.java]: Consider renaming this.

**Pull request merged**